roles/kubernetes/control-plane/defaults/main/main.yml

```diff
@@ -89,6 +89,22 @@ kube_apiserver_pod_eviction_unreachable_timeout_seconds: "300"
 # 1.10+ admission plugins
 kube_apiserver_enable_admission_plugins: []
 
+# enable admission plugins configuration
+admission_control_config_file: false
+
+# list of admission plugins that needs to be configured
+admission_plugins_needs_configuration: [EventRateLimit]
```

Comment:
New options are related to the existing `kube_apiserver_enable_admission_plugins`.
To understand it easily, how about adding the prefix `kube_apiserver_` for the consistency?
For example, this could be `kube_apiserver_admission_plugins_configured` or so on.

In addition, `admission_control_config_file` could be removed if the default value of this option is empty.

Reply:
I agree on renaming variable.

For the second one I'm not completely sure. Removing the variable form default values will generate an error in case it's not set. Isn't it?

Reply:
If the default value of this option(before renaming) is empty like
```
# list of admission plugins that needs to be configured. EventRateLimit is only supported at this time.
admission_plugins_needs_configuration: []
```
we can remove `admission_control_config_file` option by replacing with `admission_plugins_needs_configuration` like
```
- name: kubeadm | Create directory to store admission control configurations
  file:
    path: "{{ kube_config_dir }}/admission-controls"
    state: directory
    mode: 0640
  when: admission_plugins_needs_configuration|length>0
```

Reply:
Actually the variable `admission_plugins_needs_configuration` should never be empty. This is a kind of constant that is used to check if the admission plugin is configurable or not. In case someone wants to add another configurable plugin, this should be added to this list. So I can't understand why this should be empty.
If you want we can schedule a meeting to talk about this. It's a bit hard to explain this asynchronously.

Reply:
Thank you for your reply.

> In case someone wants to add another configurable plugin, this should be added to this list.

I think the above `someone` means a Kubespray contributor, not a Kubespray user, right?
Then I got your point.

I did think `admission_plugins_needs_configuration` is one of configuration options and Kubespray users should be able to change it because that is defined in roles/kubernetes/control-plane/defaults/main/main.yml where holds default values of configuration options.
I'd like to make some points clear with
- Aren't there any scenarios Kubespray users want/need to change the value of `admission_plugins_needs_configuration`? 
- If there is not any such scenario, it is better to clarify `admission_plugins_needs_configuration` is constant and Kubespray users should not change it.

Reply:
Yes, for `someone` I meant a contributor.
At this time I can't find any scenarios where the user needs to change it at "runtime".
How would you want to clarify that user doesn't have to change it?
I can suggest to move the variable under the `vars/main.yaml`, that is for role internal use only.
Otherwise we can put a comment above the variable and avoid documenting it.

Reply:
@oomichi any suggestions for the implementation?

Reply:
I tried to solve the problem. Let me know if you agree :)

Reply:
@alegrey91  Sorry for my late reply.

> I can suggest to move the variable under the vars/main.yaml, that is for role internal use only.

That seems a good idea for me.

Reply:
Ok! Code is already updated with this. Do I have to change something else?